packages/loader/container-loader/src/container.ts

```diff
@@ -833,10 +833,31 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._resolvedUrl = resolvedUrl;
             await this.connectStorageService();
 
-            // upload blobs here (NYI)
-
-            // post summary here
             if (hasAttachmentBlobs) {
+                // upload blobs to storage
+                assert(!!this.loader.services.detachedBlobStorage, "assertion for type narrowing");
+                const redirectTable: Map<string, string | undefined>
+                    = new Map(this.loader.services.detachedBlobStorage.all().map((id) => [id, undefined]));
+                do {
+                    for (const id of this.loader.services.detachedBlobStorage.all()) {
+                        if (!redirectTable.has(id)) {
+                            redirectTable.set(id, undefined);
+                        }
+                    }
+                    for (const [localId, storageId] of redirectTable) {
+                        if (storageId === undefined) {
+                            const blob = await this.loader.services.detachedBlobStorage.readBlob(localId);
+                            const response = await this.storageService.createBlob(blob);
+                            redirectTable.set(localId, response.id);
+                        }
+                    }
+                // if there are any new blobs, upload them too
+                } while (redirectTable.size < this.loader.services.detachedBlobStorage.size);
+
+                // set redirectTable on ContainerRuntime/BlobManager
+                this.context.setBlobRedirectTable(redirectTable as Map<string, string>);
```

Comment:
I do not have good suggestion, but we should brainstorm if there is a way to remove the need to have this API
Maybe by moving some of the logic here to container runtime (unlikely as it would add dependency on something else)? Or maybe this mapping should stay at loader level and loader writes out mapping blob on each summary?
It just looks weird from contract perspective, especially if scenarios does not create attachment blobs in detached container.

Reply:
Alternatively it should be a payload on createSummary() function. That seems a bit more logical IMHO (though not sure if createSummary() is called outside of attachment flow)